src/components/timeline/TrackScreenshots.js

```diff
@@ -0,0 +1,261 @@
+
+// @flow
+
+import React, { PureComponent } from 'react';
+import { createPortal } from 'react-dom';
+import explicitConnect from '../../utils/connect';
+import {
+  selectorsForThread,
+  getCommittedRange,
+  getPreviewSelection,
+} from '../../reducers/profile-view';
+import { withSize, type SizeProps } from '../shared/WithSize';
+
+import type {
+  ThreadIndex,
+  Thread,
+  MarkersTableWithPayload,
+} from '../../types/profile';
+import type { ScreenshotPayload } from '../../types/markers';
+import type { Milliseconds } from '../../types/units';
+import type {
+  ExplicitConnectOptions,
+  ConnectedProps,
+} from '../../utils/connect';
+
+import { ensureExists } from '../../utils/flow';
+import './TrackScreenshots.css';
+
+type OwnProps = {|
+  +threadIndex: ThreadIndex,
+  +screenshotId: string,
+  +overlayElement?: HTMLElement,
+  ...SizeProps,
+|};
+type StateProps = {|
+  +thread: Thread,
+  +rangeStart: Milliseconds,
+  +rangeEnd: Milliseconds,
+  +screenshots: MarkersTableWithPayload<ScreenshotPayload>,
+  +threadName: string,
+  +isMakingPreviewSelection: boolean,
+  +overlayElement: HTMLElement,
+|};
+type DispatchProps = {||};
+type Props = ConnectedProps<OwnProps, StateProps, DispatchProps>;
+type State = {|
+  offsetX: null | number,
+  pageX: null | number,
+  containerTop: null | number,
+|};
+
+export const TRACK_HEIGHT = 50;
+const HOVER_HEIGHT = 100;
+const HOVER_MAX_WIDTH_RATIO = 1.75;
+
+class Screenshots extends PureComponent<Props, State> {
+  state = {
+    offsetX: null,
+    pageX: null,
+    containerTop: null,
+  };
+
+  findScreenshotAtMouse(offsetX: number): number | null {
+    const { width, rangeStart, rangeEnd, screenshots } = this.props;
+    const rangeLength = rangeEnd - rangeStart;
+    const mouseTime = offsetX / width * rangeLength + rangeStart;
+    if (screenshots.length === 0) {
+      return null;
+    }
+    if (mouseTime < screenshots.time[0]) {
+      // Only show a screenshot for the first time we know of.
+      return null;
+    }
+    for (let i = 0; i < screenshots.length; i++) {
+      const time = screenshots.time[i];
+      if (time >= mouseTime) {
+        return i;
+      }
+    }
+    return screenshots.length - 1;
+  }
+
+  /**
+   * This function runs through all of the screenshots, and then samples the last known
+   * screenshot, and places it on the screen, making a film strip.
+   */
+  renderScreenshotStrip() {
+    const {
+      thread,
+      width: outerContainerWidth,
+      rangeStart,
+      rangeEnd,
+      screenshots,
+    } = this.props;
+    if (screenshots.length === 0) {
+      return null;
+    }
+    const images = [];
+    const rangeLength = rangeEnd - rangeStart;
+    const imageContainerWidth = TRACK_HEIGHT * 0.75;
+    const timeToPixel = time =>
+      outerContainerWidth * (time - rangeStart) / rangeLength;
+
+    let screenshotIndex = 0;
+    for (
+      let left = timeToPixel(screenshots.time[0]);
```

Comment:
if you follow my suggestion of not range filtering the screenshot data (see my comment in the reducers file), then you should likely start at `0` here.